File: app-server/src/main/java/eu/automateeverything/rest/SlowDownInterceptor.java
package eu.automateeverything.rest;

import jakarta.ws.rs.WebApplicationException;
import jakarta.ws.rs.ext.*;
import java.io.IOException;

public class SlowDownInterceptor implements WriterInterceptor, ReaderInterceptor {

    @Override
    public Object aroundReadFrom(ReaderInterceptorContext context)
            throws IOException, WebApplicationException {
        return context.proceed();
    }

    @Override
    public void aroundWriteTo(WriterInterceptorContext context)
            throws IOException, WebApplicationException {
        try {
            Thread.sleep(5000);
        } catch (InterruptedException ignored) {
        }
        context.proceed();
    }
}

File: app-server/src/main/java/eu/automateeverything/rest/SlowDownInterceptor.kt
package eu.automateeverything.rest

import jakarta.ws.rs.ext.WriterInterceptor
import jakarta.ws.rs.ext.ReaderInterceptor
import kotlin.Throws
import java.io.IOException
import jakarta.ws.rs.WebApplicationException
import jakarta.ws.rs.ext.ReaderInterceptorContext
import jakarta.ws.rs.ext.WriterInterceptorContext
import java.lang.InterruptedException

class SlowDownInterceptor : WriterInterceptor, ReaderInterceptor {
    @Throws(IOException::class, WebApplicationException::class)
    override fun aroundReadFrom(context: ReaderInterceptorContext): Any {
        return context.proceed()
    }

    @Throws(IOException::class, WebApplicationException::class)
    override fun aroundWriteTo(context: WriterInterceptorContext) {
        try {
            Thread.sleep(5000)
        } catch (ignored: InterruptedException) {
        }
        context.proceed()
    }
}